CDEFramework/Libraries/LegacyDictionarySupport/GlossaryAPIClient.cs

```diff
@@ -0,0 +1,152 @@
+
+        private HttpClient _client = null;
+
+        /// <summary>
+        /// Creates a new instance of a Glossary API client.
+        /// </summary>
+        /// <param name="client">An HttpClient that has the BaseAddress set to the API address.</param>
+        public GlossaryAPIClient()
+        {
+            string baseApiPath = WebAPISection.GetAPIUrl();
+            string appPath = ConfigurationManager.AppSettings["GlossaryAPIAppPath"];
+            string versionPath = ConfigurationManager.AppSettings["GlossaryAPIVersionPath"];
+
+            if (string.IsNullOrWhiteSpace(appPath))
+                throw new ConfigurationErrorsException("error: GlossaryAPIAppPath cannot be null or empty");
+
+            if (string.IsNullOrWhiteSpace(versionPath))
+                throw new ConfigurationErrorsException("error: GlossaryAPIVersionPath cannot be null or empty");
+
+            HttpClient client = new HttpClient();
+            //NOTE: the base URL MUST have a trailing slash
+            client.BaseAddress = new Uri(String.Format("{0}/{1}/{2}/", baseApiPath, appPath, versionPath));
+
+            this._client = client;
+        }
+
+        /// <summary>
+        /// Calls the GetById endpoint of the Glossary API
+        /// </summary>
+        /// <param name="dictionary">Dictionary name (required)</param>
+        /// <param name="audience">Audience type(required)</param>
+        /// <param name="language">Language to use (required)</param>
+        /// <param name="id">The ID of the term</param>
+        /// <param name="useFallback">Whether or not to use the fallback logic</param>
+        /// <returns>A Best Bet result</returns>
+        public GlossaryTerm GetById(
+            string dictionary,
+            string audience,
+            string language,
+            string id,
+            bool useFallback = false 
+            )
+        {
+            //GlossaryTerm resultTerm = null;
+
+            // Check fields
+            if (String.IsNullOrWhiteSpace(dictionary))
+            {
+                throw new ArgumentNullException("The dictionary is null or an empty string");
+            }
+            if (String.IsNullOrWhiteSpace(audience))
+            {
+                throw new ArgumentNullException("The audience is null or an empty string");
+            }
+            if (String.IsNullOrWhiteSpace(language))
+            {
+                throw new ArgumentNullException("The language is null or an empty string");
+            }
+            if (String.IsNullOrWhiteSpace(id))
+            {
+                throw new ArgumentNullException("The ID is null or an empty string");
+            }
+
+            // Set up search param string: {dictionary}/{audience}/{language}/{id:long}
+            string[] searchParams = { dictionary, audience, language, id, useFallback.ToString() };
+            string searchParam = string.Format("{0}/{1}/{2}/{3}?useFallback={4}", searchParams);
```

Comment:
The positional parameters can be eliminated and readability/maintainability improved by writing this as:

```csharp
string searchParam = $"{dictionary}/{audience}/{language}/{id}?useFallback={useFallback}";
```